### PR TITLE
SAIC-772 Added tests for server groups

### PR DESCRIPTION
**Pull request by ntymtsiv**

### CODE REVIEW

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.utils
devlab/tests/utils.py:17: [W0403(relative-import), ] Relative import 'config', should be 'tests.config'
devlab/tests/utils.py:96: [W0621(redefined-outer-name), MigrationUtils.__init__] Redefining name 'config' from outer scope (line 17)
devlab/tests/utils.py:107: [E1129(not-context-manager), MigrationUtils.execute_command_on_vm] Context manager 'generator' doesn't implement __enter__ and __exit__.
************* Module tests.test_resource_migration
devlab/tests/test_resource_migration.py:1: [W0403(relative-import), ] Relative import 'config', should be 'tests.config'
devlab/tests/test_resource_migration.py:2: [W0403(relative-import), ] Relative import 'test_exceptions', should be 'tests.test_exceptions'
devlab/tests/test_resource_migration.py:3: [W0403(relative-import), ] Relative import 'functional_test', should be 'tests.functional_test'
devlab/tests/test_resource_migration.py:425: [W0212(protected-access), ResourceMigrationTests.test_migrate_tenant_quotas.get_tenant_quotas] Access to a protected member _info of a client class
devlab/tests/test_resource_migration.py:435: [W0212(protected-access), ResourceMigrationTests.test_migrate_tenant_quotas] Access to a protected member _info of a client class
************* Module tests.generate_load
devlab/tests/generate_load.py:8: [W0403(relative-import), ] Relative import 'config', should be 'tests.config'
devlab/tests/generate_load.py:9: [W0403(relative-import), ] Relative import 'utils', should be 'tests.utils'
devlab/tests/generate_load.py:23: [W0403(relative-import), ] Relative import 'test_exceptions', should be 'tests.test_exceptions'
devlab/tests/generate_load.py:768: [E1101(no-member), Prerequisites.create_security_group] Instance of 'dict' has no 'id' member
devlab/tests/generate_load.py:842: [E1101(no-member), Prerequisites.create_cinder_volumes] Instance of 'dict' has no 'id' member

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.utils
devlab/tests/utils.py:17: [W0403(relative-import), ] Relative import 'config', should be 'tests.config'
devlab/tests/utils.py:96: [W0621(redefined-outer-name), MigrationUtils.__init__] Redefining name 'config' from outer scope (line 17)
devlab/tests/utils.py:107: [E1129(not-context-manager), MigrationUtils.execute_command_on_vm] Context manager 'generator' doesn't implement __enter__ and __exit__.
************* Module tests.test_resource_migration
devlab/tests/test_resource_migration.py:1: [W0403(relative-import), ] Relative import 'config', should be 'tests.config'
devlab/tests/test_resource_migration.py:2: [W0403(relative-import), ] Relative import 'test_exceptions', should be 'tests.test_exceptions'
devlab/tests/test_resource_migration.py:3: [W0403(relative-import), ] Relative import 'functional_test', should be 'tests.functional_test'
devlab/tests/test_resource_migration.py:425: [W0212(protected-access), ResourceMigrationTests.test_migrate_tenant_quotas.get_tenant_quotas] Access to a protected member _info of a client class
devlab/tests/test_resource_migration.py:435: [W0212(protected-access), ResourceMigrationTests.test_migrate_tenant_quotas] Access to a protected member _info of a client class
************* Module tests.generate_load
devlab/tests/generate_load.py:8: [W0403(relative-import), ] Relative import 'config', should be 'tests.config'
devlab/tests/generate_load.py:9: [W0403(relative-import), ] Relative import 'utils', should be 'tests.utils'
devlab/tests/generate_load.py:23: [W0403(relative-import), ] Relative import 'test_exceptions', should be 'tests.test_exceptions'
devlab/tests/generate_load.py:770: [E1101(no-member), Prerequisites.create_security_group] Instance of 'dict' has no 'id' member
devlab/tests/generate_load.py:844: [E1101(no-member), Prerequisites.create_cinder_volumes] Instance of 'dict' has no 'id' member

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.utils
devlab/tests/utils.py:17: [W0403(relative-import), ] Relative import 'config', should be 'tests.config'
devlab/tests/utils.py:96: [W0621(redefined-outer-name), MigrationUtils.__init__] Redefining name 'config' from outer scope (line 17)
devlab/tests/utils.py:107: [E1129(not-context-manager), MigrationUtils.execute_command_on_vm] Context manager 'generator' doesn't implement __enter__ and __exit__.
************* Module tests.test_resource_migration
devlab/tests/test_resource_migration.py:15: [W0403(relative-import), ] Relative import 'config', should be 'tests.config'
devlab/tests/test_resource_migration.py:16: [W0403(relative-import), ] Relative import 'test_exceptions', should be 'tests.test_exceptions'
devlab/tests/test_resource_migration.py:17: [W0403(relative-import), ] Relative import 'functional_test', should be 'tests.functional_test'
devlab/tests/test_resource_migration.py:456: [W0212(protected-access), ResourceMigrationTests.test_migrate_tenant_quotas.get_tenant_quotas] Access to a protected member _info of a client class
devlab/tests/test_resource_migration.py:466: [W0212(protected-access), ResourceMigrationTests.test_migrate_tenant_quotas] Access to a protected member _info of a client class
************* Module tests.generate_load
devlab/tests/generate_load.py:8: [W0403(relative-import), ] Relative import 'config', should be 'tests.config'
devlab/tests/generate_load.py:9: [W0403(relative-import), ] Relative import 'utils', should be 'tests.utils'
devlab/tests/generate_load.py:23: [W0403(relative-import), ] Relative import 'test_exceptions', should be 'tests.test_exceptions'
devlab/tests/generate_load.py:827: [E1101(no-member), Prerequisites.create_security_group] Instance of 'dict' has no 'id' member
devlab/tests/generate_load.py:901: [E1101(no-member), Prerequisites.create_cinder_volumes] Instance of 'dict' has no 'id' member

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by ntymtsiv** [comment]
depend on #701

File: devlab/config.template [hunk 1]
```diff
@@ -29,7 +29,7 @@ cinder_migration_strategy = cloudferrylib.os.storage.cinder_database.CinderStora
 migrate_user_quotas = False
 incloud_live_migration = nova
 ssh_connection_attempts = 3
-keep_affinity_settings = False
+keep_affinity_settings = True
```
Comment (plomakin) [inline devlab/config.template]: Is this changes related?

**Comment by plomakin** [comment]
retest this please

**Comment by plomakin** [comment]
+1 looks good. lets get at least one build passed successfully

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.utils
devlab/tests/utils.py:17: [W0403(relative-import), ] Relative import 'config', should be 'tests.config'
devlab/tests/utils.py:122: [E1129(not-context-manager), MigrationUtils.execute_command_on_vm] Context manager 'generator' doesn't implement __enter__ and __exit__.
************* Module tests.test_resource_migration
devlab/tests/test_resource_migration.py:15: [W0403(relative-import), ] Relative import 'config', should be 'tests.config'
devlab/tests/test_resource_migration.py:16: [W0403(relative-import), ] Relative import 'test_exceptions', should be 'tests.test_exceptions'
devlab/tests/test_resource_migration.py:17: [W0403(relative-import), ] Relative import 'functional_test', should be 'tests.functional_test'
devlab/tests/test_resource_migration.py:510: [W0212(protected-access), ResourceMigrationTests.test_migrate_tenant_quotas.get_tenant_quotas] Access to a protected member _info of a client class
devlab/tests/test_resource_migration.py:520: [W0212(protected-access), ResourceMigrationTests.test_migrate_tenant_quotas] Access to a protected member _info of a client class
************* Module tests.generate_load
devlab/tests/generate_load.py:7: [W0403(relative-import), ] Relative import 'config', should be 'tests.config'
devlab/tests/generate_load.py:8: [W0403(relative-import), ] Relative import 'utils', should be 'tests.utils'
devlab/tests/generate_load.py:21: [W0403(relative-import), ] Relative import 'test_exceptions', should be 'tests.test_exceptions'
devlab/tests/generate_load.py:825: [E1101(no-member), Prerequisites.create_security_group] Instance of 'dict' has no 'id' member
devlab/tests/generate_load.py:899: [E1101(no-member), Prerequisites.create_cinder_volumes] Instance of 'dict' has no 'id' member

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by plomakin** [comment]
can be merged

**Comment by roman-verchikov** [comment]
retest this please

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.utils
devlab/tests/utils.py:18: [W0403(relative-import), ] Relative import 'config', should be 'tests.config'
devlab/tests/utils.py:35: [W0141(bad-builtin), convert] Used builtin function 'map'. Using a list comprehension can be clearer.
devlab/tests/utils.py:37: [W0141(bad-builtin), convert] Used builtin function 'map'. Using a list comprehension can be clearer.
devlab/tests/utils.py:135: [E1129(not-context-manager), MigrationUtils.execute_command_on_vm] Context manager 'generator' doesn't implement __enter__ and __exit__.
************* Module tests.test_resource_migration
devlab/tests/test_resource_migration.py:15: [W0403(relative-import), ] Relative import 'config', should be 'tests.config'
devlab/tests/test_resource_migration.py:16: [W0403(relative-import), ] Relative import 'test_exceptions', should be 'tests.test_exceptions'
devlab/tests/test_resource_migration.py:17: [W0403(relative-import), ] Relative import 'functional_test', should be 'tests.functional_test'
devlab/tests/test_resource_migration.py:550: [W0212(protected-access), ResourceMigrationTests.test_migrate_tenant_quotas.get_tenant_quotas] Access to a protected member _info of a client class
devlab/tests/test_resource_migration.py:560: [W0212(protected-access), ResourceMigrationTests.test_migrate_tenant_quotas] Access to a protected member _info of a client class
************* Module tests.generate_load
devlab/tests/generate_load.py:7: [W0403(relative-import), ] Relative import 'config', should be 'tests.config'
devlab/tests/generate_load.py:8: [W0403(relative-import), ] Relative import 'utils', should be 'tests.utils'
devlab/tests/generate_load.py:21: [W0403(relative-import), ] Relative import 'test_exceptions', should be 'tests.test_exceptions'
devlab/tests/generate_load.py:838: [E1101(no-member), Prerequisites.create_security_group] Instance of 'dict' has no 'id' member
devlab/tests/generate_load.py:912: [E1101(no-member), Prerequisites.create_cinder_volumes] Instance of 'dict' has no 'id' member

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by roman-verchikov** [comment]
+1, can be merged
